File: 119. Pascal's Triangle II/119. Pascal's Triangle II.rb
def get_row(row_index)
    row = Array.new(row_index + 1, 1)
    
    (1..row_index).each do |i|
        prev = row[0]
        (1...i).each do |j|
            temp = row[j]
            row[j] = (prev * (row_index - j + 1)) / j
            prev = temp
        end
        row[i] = 1
    end
    
    row
end

File: 119. Pascal's Triangle II/119. Pascal's Triangle II.ts
function getRow(rowIndex: number): number[] {
    const row: number[] = new Array(rowIndex + 1).fill(1);
    
    for (let i = 1; i <= rowIndex; i++) {
        let prev: number = row[0];
        for (let j = 1; j < i; j++) {
            const temp = row[j];
            // Use multiplication before division to maintain precision
            row[j] = Math.floor((prev * (rowIndex - j + 1)) / j);
            prev = temp;
        }
        row[i] = 1;
    }
    
    return row;
}